pkg/apis/workflow/v1alpha1/workflow_types_test.go

```diff
@@ -79,6 +79,23 @@ func TestWorkflowHappenedBetween(t *testing.T) {
 	}))
 }
 
+func TestArtifact_ValidatePath(t *testing.T) {
+	t.Run("empty path fails", func(t *testing.T) {
+		err := Artifact{Name: "a1", Path: ""}.ValidatePath()
+		assert.EqualError(t, err, "Artifact a1 did not specify a path")
+	})
+
+	t.Run("directory traversal fails", func(t *testing.T) {
+		err := Artifact{Name: "a1", Path: "../../etc/passwd"}.ValidatePath()
+		assert.Contains(t, err.Error(), "Directory traversal is not permitted")
+	})
+
+	t.Run("normal artifact path succeeds", func(t *testing.T) {
+		err := Artifact{Name: "a1", Path: "/tmp/some-artifact.txt"}.ValidatePath()
+		assert.NoError(t, err)
+	})
+}
```

Comment:
Add a test to make sure it allows filenames with embedded `..`.

Reply:
Added in https://github.com/argoproj/argo-workflows/pull/7187/commits/6fed6857743cd4abeef3951fe4e34485162ea5f3